Sources/Logging/Log+DefaultLogger.swift

```diff
@@ -0,0 +1,204 @@
+        ///
+        /// - Parameters:
+        ///   - module: The module to be registered.
+        ///   - minLevel: The minimum severity level required to be logged by the module.
+        /// - Throws: A `DefaultLoggerError.duplicateModule` error if a module with the same `rawValue` is already
+        /// registered.
+        public func registerModule(_ module: Module, minLevel: Level) throws {
+
+            try modules.modify {
+                guard $0[module] == nil else { throw DefaultLoggerError.duplicateModule(module.rawValue) }
+
+                $0[module] = minLevel
+            }
+        }
+
+        /// Unregisters a module from the logger, taking it into account when filtering any new log messages (if logged
+        /// with a non `nil` module)
+        ///
+        /// - SeeAlso: `registerModule(_:minLevel:)`
+        ///
+        /// - Parameter module: The module to be unregistered.
+        /// - Throws: A `DefaultLoggerError.inexistentModule` error if a module with the same `rawValue` isn't
+        /// registered.
+        public func unregisterModule(_ module: Module) throws {
+
+            try modules.modify {
+                guard let _ = $0.removeValue(forKey: module) else {
+                    throw DefaultLoggerError.inexistentModule(module.rawValue)
+                }
+            }
+        }
+
+        // MARK: - Logging
+
+        /// Logs a message from the specified module (if non `nil`) with the given level, alongside the file, function
+        /// and line the log originated from.
+        ///
+        /// - Note:
+        /// If the `module` parameter is non `nil`, the message will only be logged if the module is registered in the
+        /// logger, and the log message's level is *above* the module's registered minimum log level. On the other hand,
+        /// if `module` is set to `nil`, no module filtering will be applied.
+        ///
+        /// - Parameters:
+        ///   - module: The module from which the message originated. Set to `nil` for no module filtering.
+        ///   - level: The severity level of the message.
+        ///   - message: The log message.
+        ///   - file: The file from where the log was invoked.
+        ///   - line: The line from where the log was invoked.
+        ///   - function: The function from where the log was invoked.
+        public func log(module: Module?,
+                        level: Level,
+                        message: @autoclosure () -> String,
+                        file: StaticString = #file,
+                        line: UInt = #line,
+                        function: StaticString = #function) {
+
+            // skip module checks for `nil` modules
+            if let module = module {
+                guard
+                    let moduleMinLevel = modules.value[module],
+                    level.isAbove(minLevel: moduleMinLevel)
+                    else { return }
```

Comment:
Wrong indentation.

Reply:
Well spotted, damn Xcode and its opinionated auto formatting 😓